backend/pkg/logger/data/logger.go

```diff
@@ -0,0 +1,115 @@
+package data
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/logger"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/packet/data"
+)
+
+type ValueName string
+
+const (
+	name abstraction.LoggerName = "data"
+)
+
+type Logger struct {
+	running    bool
+	lock       sync.Mutex
```

Comment:
From what I am seeing you only use the lock on the start and stop methods to protect the running variable.
If that is the only place where you intend to use it, make `running` be an [atomic boolean](https://pkg.go.dev/sync/atomic#Bool) (atomic values are already synchronized).
If you also intend to protect the map in the `Push` and `Pull` functions, use a [sync map](https://pkg.go.dev/sync#Map) or [read-write mutex](https://pkg.go.dev/sync#RWMutex). You can read more about other synchronization methods [here](https://hjr265.me/blog/synchronization-constructs-in-go-standard-library/)